src/app/chat/view.nim

```diff
@@ -547,3 +579,169 @@ QtObject:
+    if (community.id == ""):
+      discard self.getCommunitiesIfNotFetched()
+      community = self.communityList.getCommunityById(communityId) 
+    self.observedCommunity.setCommunityItem(community)
+    self.observedCommunityChanged()
+
+  proc getObservedCommunity*(self: ChatsView): QVariant {.slot.} =
+    newQVariant(self.observedCommunity)
+
+  QtProperty[QVariant] observedCommunity:
+    read = getObservedCommunity
+    write = setObservedCommunity
+    notify = observedCommunityChanged
+
+  proc joinCommunity*(self: ChatsView, communityId: string): string {.slot.} =
+    result = ""
+    try:
+      self.status.chat.joinCommunity(communityId)
+      self.joinedCommunityList.addCommunityItemToList(self.communityList.getCommunityById(communityId))
+      self.setActiveCommunity(communityId)
+    except Exception as e:
+      error "Error joining the community", msg = e.msg
+      result = fmt"Error joining the community: {e.msg}"
+
+  proc leaveCommunity*(self: ChatsView, communityId: string): string {.slot.} =
+    result = ""
+    try:
+      self.status.chat.leaveCommunity(communityId)
+      if (communityId == self.activeCommunity.communityItem.id):
+        self.activeCommunity.setActive(false)
+      self.joinedCommunityList.removeCommunityItemFromList(communityId)
+      var updatedCommunity = self.communityList.getCommunityById(communityId)
+      updatedCommunity.joined = false
+      self.communityList.replaceCommunity(updatedCommunity)
+    except Exception as e:
+      error "Error leaving the community", msg = e.msg
+      result = fmt"Error leaving the community: {e.msg}"
+
+  proc leaveCurrentCommunity*(self: ChatsView): string {.slot.} =
+    result = self.leaveCommunity(self.activeCommunity.communityItem.id)
+
+  proc inviteUserToCommunity*(self: ChatsView, pubKey: string): string {.slot.} =
+    try:
+      self.status.chat.inviteUserToCommunity(self.activeCommunity.id(), pubKey)
+    except Exception as e:
+      error "Error inviting to the community", msg = e.msg
+      result = fmt"Error inviting to the community: {e.msg}"
+
+  proc exportComumnity*(self: ChatsView): string {.slot.} =
+    try:
+      result = self.status.chat.exportCommunity(self.activeCommunity.communityItem.id)
+    except Exception as e:
+      error "Error exporting the community", msg = e.msg
+      result = fmt"Error exporting the community: {e.msg}"
+
+  proc importCommunity*(self: ChatsView, communityKey: string) {.slot.} =
+    try:
+      self.status.chat.importCommunity(communityKey)
+    except Exception as e:
+      error "Error importing the community", msg = e.msg
+
+  proc removeUserFromCommunity*(self: ChatsView, pubKey: string) {.slot.} =
+    try:
+      self.status.chat.removeUserFromCommunity(self.activeCommunity.id(), pubKey)
+      self.activeCommunity.removeMember(pubKey)
+    except Exception as e:
+      error "Error removing user from the community", msg = e.msg
```

Comment:
We should probably extract the community code to a separate QObject. But this can be done in a separate refactor PR